src/DeviceManager.cpp

```diff
@@ -143,7 +143,7 @@ static void AddSourcesFromStream(int deviceIndex, const PaDeviceInfo *info, std:
    FillHostDeviceInfo(&map, info, deviceIndex, 1);
 
 #ifdef USE_PORTMIXER
-   PxMixer *portMixer = Px_OpenMixer(stream, 0);
+   PxMixer *portMixer = Px_OpenMixer(stream, deviceIndex, -1, 0);
```

Comment:
I stepped through this function and thought that all the complementary Px_CloseMixer calls would be nicer if done by RAII.  Put this somewhere in the C++ headers outside of PortMixer:

```
struct PortMixerDeleter{ void operator ()(void *p) { if (p) Px_CloseMixer(p); } };
using PortMixerHandle = std::unique_ptr<PxMixer, PortMixerDeleter>;
```

But that cleanup is beside the point of this PR of course.

Reply:
I also now notice `#ifdef USE_PORTMIXER` in some places but `#if USE_PORTMIXER` in others, and I don't know if the distinction was intended.  Are builds with USE_PORTMIXER defined but zero really supposed to be different?

Another cleanup perhaps not for the scope of this issue.